### PR TITLE
Viber public channel

### PR DESCRIPTION
**Pull request by nicpottier**

### CODE REVIEW

File: temba/channels/handlers.py [hunk 1]
```diff
@@ -2088,3 +2089,210 @@ def post(self, request, *args, **kwargs):
...
+        return HttpResponse("Must be called as a POST", status=405)
+
+    def post(self, request, *args, **kwargs):
+        from temba.msgs.models import Msg
+        request_uuid = kwargs['uuid']
+
+        # look up the channel
+        channel = Channel.objects.filter(uuid=request_uuid, is_active=True, channel_type=Channel.TYPE_VIBER_PUBLIC).exclude(org=None).first()
+        if not channel:
+            return HttpResponse("Channel not found for id: %s" % request_uuid, status=400)
```
Comment (ericnewcomer) [inline temba/channels/handlers.py]: What does Viber do if it receives a 400? Does it retry?

File: temba/channels/models.py [hunk 2]
```diff
@@ -323,6 +326,45 @@ def add_viber_channel(cls, org, user, name):
...
+                raise Exception(_("Unable to set webhook for channel: %s", response.text))
+
+            response_json = response.json()
+            if response_json['status'] != 0:  # pragma: no cover
+                raise Exception(_("Unable to set Viber webhook: %s" % response_json['status_message']))
+
+            return channel
+
+        except Exception as e:
+            raise e
```
Comment (ericnewcomer) [inline temba/channels/models.py]: purpose here?

File: temba/channels/handlers.py (hunk 1 above)
Comment (nicpottier) [inline temba/channels/handlers.py]: Looks like it retries for anything non-200, so I guess 200 is right? seems wrong but better than them retrying I suppose.

File: temba/channels/models.py (hunk 2 above)
Comment (nicpottier) [inline temba/channels/models.py]: Ah, right, ya, not so useful.

**Comment by ericnewcomer** [comment]
There's lots of other places we are returning 400s too (mostly with them sending invalid requests for one reason or another). Should we maybe absorb these too with an error message and a 200? t'would be a shame for them to keep hammering us with badness.

**Comment by nicpottier** [comment]
Seems like if they send us an invalid request we want they to retry with a valid one. They may have bugs too. I'm ok changing that later if we see it become an issue but I'd rather not fail silently as a start.

**Comment by nicpottier** [comment]
purty please?

**Comment by ericnewcomer** [comment]
Oops, forgot to reapprove!

**Pull request merged**